models/node.js

```diff
@@ -1,14 +1,40 @@
 import { formatPercent } from '@/utils/string';
+import { NODE_ROLES } from '@/config/labels-annotations.js';
 
 export default {
   name() {
     return this.metadata.name;
   },
 
   roles() {
-    console.warn('no backend for roles');
-
-    return 'All';
+    const {
+      CONTROL_PLANE: controlPlane,
+      WORKER:        worker,
+      ETCD:          etcd
+    } = NODE_ROLES;
+    const labels = this.metadata?.labels;
+
+    const isControlPlane = labels[controlPlane];
+    const isWorker = labels[worker];
+    const isEtcd = labels[etcd];
+
+    if (( isControlPlane && isWorker && isEtcd ) ||
+        ( !isControlPlane && !isWorker && !isEtcd )) {
+      // !isControlPlane && !isWorker && !isEtcd === RKE?
+      return 'All';
+    }
+
+    if (isControlPlane) {
```

Comment:
All 8 combinations are possible, you're only handling 5 of them.  (worker+cp, worker+etcd, cp+etcd)